src/main/vcph7/usbd_cdc_hid.c

```diff
@@ -0,0 +1,336 @@
+
+__ALIGN_BEGIN static uint8_t USBD_HID_CDC_CfgDesc[USB_HID_CDC_CONFIG_DESC_SIZ] __ALIGN_END =
+{
+  0x09,                                    /* bLength: Configuration Descriptor size */
+  USB_DESC_TYPE_CONFIGURATION,       /* bDescriptorType: Configuration */
+  USB_HID_CDC_CONFIG_DESC_SIZ,
+  /* wTotalLength: Bytes returned */
+  0x00,
+  0x03,                                    /*bNumInterfaces: 2 interfaces (1 for CDC, 1 for HID)*/
+  0x01,                                    /*bConfigurationValue: Configuration value*/
+  0x00,                                    /*iConfiguration: Index of string descriptor describing
+                                              the configuration*/
+  0xC0,                                    /*bmAttributes: bus powered and Support Remote Wake-up */
+  0x32,                                    /*MaxPower 100 mA: this current is used for detecting Vbus*/
+
+  /************** Descriptor of Joystick Mouse interface ****************/
+  /* 09 */
+  0x09,                                    /*bLength: Interface Descriptor size*/
+  USB_DESC_TYPE_INTERFACE,           /*bDescriptorType: Interface descriptor type*/
+  HID_INTERFACE,                           /*bInterfaceNumber: Number of Interface*/
+  0x00,                                    /*bAlternateSetting: Alternate setting*/
+  0x01,                                    /*bNumEndpoints*/
+  0x03,                                    /*bInterfaceClass: HID*/
+  0x00,                                    /*bInterfaceSubClass : 1=BOOT, 0=no boot*/
+  0x00,                                    /*nInterfaceProtocol : 0=none, 1=keyboard, 2=mouse*/
+  0,                                       /*iInterface: Index of string descriptor*/
+  /******************** Descriptor of Joystick Mouse HID ********************/
+  /* 18 */
+  0x09,                                    /*bLength: HID Descriptor size*/
+  HID_DESCRIPTOR_TYPE,                     /*bDescriptorType: HID*/
+  0x11,                                    /*bcdHID: HID Class Spec release number*/
+  0x01,
+  0x00,                                    /*bCountryCode: Hardware target country*/
+  0x01,                                    /*bNumDescriptors: Number of HID class descriptors to follow*/
+  0x22,                                    /*bDescriptorType*/
+  HID_MOUSE_REPORT_DESC_SIZE,              /*wItemLength: Total length of Report descriptor*/
+  0x00,
+  /******************** Descriptor of Mouse endpoint ********************/
+  /* 27 */
+  0x07,                                    /*bLength: Endpoint Descriptor size*/
+  USB_DESC_TYPE_ENDPOINT,            /*bDescriptorType:*/
+
+  HID_EPIN_ADDR,                           /*bEndpointAddress: Endpoint Address (IN)*/
+  0x03,                                    /*bmAttributes: Interrupt endpoint*/
+  HID_EPIN_SIZE,                           /*wMaxPacketSize: 8 Byte max */
+  0x00,
+  HID_POOLING_INTERVAL,                    /*bInterval: Polling Interval (10 ms)*/
```

Comment:
I like me a good pooling. :rofl:

Reply:
POOLING in Betaflight v.s. O_CREAT in *nix